src/core/thread/thread_netif.cpp

```diff
@@ -192,21 +192,20 @@ ThreadError ThreadNetif::RouteLookup(const Ip6::Address &source, const Ip6::Addr
     return error;
 }
 
-ThreadError ThreadNetif::TmfFilter(const Message &aMessage, const Ip6::MessageInfo &aMessageInfo)
+ThreadError ThreadNetif::TmfFilter(const Message &aMessage, const Ip6::MessageInfo &aMessageInfo, void *aContext)
 {
     ThreadError error = kThreadError_None;
 
     // A TMF message must comply one of the following rules:
-    // 1. Source address is RLOC or ALOC, and destination address is RLOC, ALOC or realm-local multicast.
-    // 2. Both source and destination addresses are link-local.(for Joiner Entrust)
+    // 1. Source address is RLOC or ALOC, and destination address is mesh local address(that is, RLOC, ALOC, or ML-ELD) or multicast.
+    // 2. Both source and destination addresses are link-local (for Joiner Entrust).
```

Comment:
Isn't this still more strict than section 10.6.2 of the specification?
suggestion:
```
(aMessageInfo.GetPeerAddr().IsRoutingLocator() || aMessageInfo.GetPeerAddr().IsAnycastRoutingLocator()) ||
(aMessageInfo.GetSockAddr().IsRoutingLocator() || aMessageInfo.GetPeerSock().IsAnycastRoutingLocator()) ||
aMessageInfo.GetSockAddr().IsLinkLocal()
```